packages/ckeditor5-engine/src/view/matcher.ts

```diff
@@ -389,27 +386,34 @@ function matchPatterns(
  */
 function normalizePatterns( patterns: PropertyPatterns ): Array<[ true | string | RegExp, true | string | RegExp ]> {
 	if ( Array.isArray( patterns ) ) {
-		return patterns.map( ( pattern: any ) => {
-			if ( isPlainObject( pattern ) ) {
-				if ( pattern.key === undefined || pattern.value === undefined ) {
-					// Documented at the end of matcher.js.
-					logWarning( 'matcher-pattern-missing-key-or-value', pattern );
-				}
+		return patterns.map( pattern => {
+			if ( typeof pattern !== 'object' || pattern instanceof RegExp ) {
```

Comment:
Wouldn't it be safer to check if it is a string?

> // Assume the pattern is either String or RegExp.

Reply:
No, this check allows for either Regexp or string, not only string.

Reply:
That's what I meant, `typeof pattern == 'string' || pattern instanceof RegExp` :smile:.

Reply:
Right, makes sense